server/commands/list_objects.go

```diff
@@ -49,7 +51,39 @@ func (q *ListObjectsQuery) Execute(ctx context.Context, req *openfgapb.ListObjec
 		return nil, serverErrors.NewInternalError("", err)
 	}
 
-	resultsChan := make(chan string, q.ListObjectsMaxResults)
+	targetObjectType := req.GetType()
+	targetRelation := req.GetRelation()
+
+	model, err := q.Datastore.ReadAuthorizationModel(ctx, req.GetStoreId(), req.GetAuthorizationModelId())
+	if err != nil {
+		if errors.Is(err, storage.ErrNotFound) {
+			return nil, serverErrors.TypeNotFound(targetObjectType)
+		}
+		return nil, err
+	}
+
+	typesys := typesystem.New(model)
+
+	_, ok := typesys.GetTypeDefinition(targetObjectType)
+	if !ok {
+		return nil, serverErrors.TypeNotFound(targetObjectType)
+	}
+
+	relation, ok := typesys.GetRelation(targetObjectType, targetRelation)
+	if !ok {
+		return nil, serverErrors.RelationNotFound(targetRelation, targetObjectType, nil)
+	}
+
+	hasTypeInfo, ok := typesys.HasTypeInfo(targetObjectType, targetRelation)
+	if !ok {
```

Comment:
I assume this is to find out the usage of the previous version? Perhaps add a comment for that context. Also, please use named instead of a single string as it's harder to search and do aggregations.

Also, I believe there is a utility `WithContext` that further enriches the log with bits such as the request ID and store ID.

Reply:
> I assume this is to find out the usage of the previous version?

Not necessarily, but it could also be used for that I suppose. We need to know if the model being used has type information because if it doesn't then we can't use the `ConnectedObjects` optimization. So this is mainly for that purpose.

> please use named instead of a single string as it's harder to search and do aggregations.

I could change to named fields with something like `{"error": "failed to lookup type info for relation", "relation": targetRelation, "objectType": targetObjectType}`, but having named fields for the relation and objectType there is odd. The only real valuable named param in this context would be the error field itself, and then the relation and objectType fields are just tag along and contextually specific to that error.

Not sure I understand the full reasoning here? Could you clarify?

Are you suggesting
```
msg := fmt.Sprintf("failed to lookup type info for relation '%s' on object type '%s'", targetRelation, targetObjectType)

q.Logger.WarnWithContext(ctx, msg, zap.Error("failed to lookup type info"))
```

> Also, I believe there is a utility WithContext that further enriches the log with bits such as the request ID and store ID.

I switched it to `Logger.WarnWithContext` 👍

Reply:
Im suggesting:
`(ctx, "failed to lookup type info", zap.String("relation", targetRelation), zap.String("type", targetObjectType))

Reply:
Ok, I'm happy to do that, but it's quite weird IMO. Relation and type have no need to be searchable fields in this context, but that's ok.

I'd be searching logs on the string starting with "failed to lookup type info" and the relation and object type would be in the log message. I don't see a need to break those out into individual fields of the log message. If it was something like `requestID` or some worth searching/indexing a field for then I'd feel differently, but not in this case.

Reply:
When performing an audit of the system I can immediately think of:
filter by: `"failed to lookup type info`, group by: `store id, type`, from the last week

That would give me all the offending stores and I can communicate them specifically which `type` needs migration/love per store. Relation would be a bit too much 😅

Reply:
Mmm, the `groupby(storeId, type)` is pretty sexy 😄 I likey 👍 

So how about we go with, for example,

```
(ctx, "failed to lookup type info for relation 'viewer'", zap.String("storeId", store), zap.String("type", targetObjectType))
```

how does that sound?